ocaml/parsing/pprintast.ml

```diff
@@ -388,6 +388,9 @@ and jkind ctxt f k = match (k : Jane_syntax.Jkind.t) with
   | With (t, ty) ->
     pp f "%a with %a" (jkind ctxt) t (core_type ctxt) ty
   | Kind_of ty -> pp f "kind_of_ %a" (core_type ctxt) ty
+  | Arrow (args, result) ->
+      pp f "(%a) => %a" (list (jkind ctxt) ~sep:", ") args (jkind ctxt) result
```

Comment:
This will print `(value) => value`. It would be better to match on cases where there is a single argument of base kind and avoid the parentheses in those cases.

Reply:
I found this printing logic to be implemented many times and was unsatisfied with essentially rewriting the same piece of logic (e.g. in #2802 this comes up in `oprint.ml` and in `jkind.ml`). I kept it simple in case there's a way of only implementing it just once. Do you think it's better if it's rewritten in many places or is there a way of doing this generically?

Reply:
I think it is fine to repeat it in three places. The repitition between here and `oprint` is normal and somewhat sensible. Presumably `jkind.ml` has its own printer for some error messages. That one should maybe be shared with `oprint`, but the cost of a little duplication is pretty low.